pkg/query-service/app/clickhouseReader/reader.go

```diff
@@ -3654,6 +3655,41 @@ func (r *ClickHouseReader) QueryDashboardVars(ctx context.Context, query string)
 	return &result, nil
 }
 
+func (r *ClickHouseReader) GetMetricAggregateAttributes(ctx context.Context, req *v3.AggregateAttributeRequest) (*v3.AggregateAttributeResponse, error) {
+
+	var query string
+	var err error
+	var rows driver.Rows
+	var response v3.AggregateAttributeResponse
+
+	query = fmt.Sprintf("SELECT DISTINCT(metric_name) from %s.%s WHERE metric_name ILIKE $1", signozMetricDBName, signozTSTableName)
+	if req.Limit != 0 {
+		query = query + fmt.Sprintf(" LIMIT %d;", req.Limit)
+	}
+	rows, err = r.db.Query(ctx, query, fmt.Sprintf("%%%s%%", req.SearchText))
+
+	if err != nil {
+		zap.S().Error(err)
+		return nil, fmt.Errorf("error while executing query: %s", err.Error())
+	}
+	defer rows.Close()
+
+	var metricName string
+	for rows.Next() {
+		if err := rows.Scan(&metricName); err != nil {
+			return nil, fmt.Errorf("error while scanning rows: %s", err.Error())
+		}
+		key := v3.AttributeKey{
+			Key:      metricName,
+			DataType: "Float",
```

Comment:
I wasn't really sure about this here. This could be as detailed as a bit included exact type from the DB or simply the number. The client doesn't need to bother about that because it's all `number`s for them. Used the constants defined earlier for data type but added the new constants for type. What do you think about the data type? Do you see a case where it needs to really explicit than simple "number"?

Reply:
For logs and traces we are going with simple "number" . I think that should be good.

Reply:
It's fine for the metrics as well.

Reply:
Cool, let's go with three constants then `string`, `number`,  `bool` . cc @makeavish